src/data/systemIntake.ts

```diff
@@ -2,48 +2,48 @@ import {
   GovernanceCollaborationTeam,
   SystemIntakeForm
 } from 'types/systemIntake';
+import { v4 as uuidv4 } from 'uuid';
 import cmsGovernanceTeams from '../constants/enums/cmsGovernanceTeams';
 
-export const initialSystemIntakeForm: SystemIntakeForm = {
-  id: '',
-  euaUserID: '',
-  projectName: '',
-  status: 'DRAFT',
-  requester: {
-    name: '',
-    component: ''
-  },
-  businessOwner: {
-    name: '',
-    component: ''
-  },
-  productManager: {
-    name: '',
-    component: ''
-  },
-  isso: {
-    isPresent: null,
-    name: ''
-  },
-  governanceTeams: {
-    isPresent: null,
-    teams: []
-  },
-  fundingSource: {
-    isFunded: null,
-    fundingNumber: ''
-  },
-  businessNeed: '',
-  businessSolution: '',
-  currentStage: '',
-  needsEaSupport: null,
-  hasContract: ''
+export const initialSystemIntakeForm = (): SystemIntakeForm => {
+  return {
+    id: uuidv4(),
```

Comment:
Can you explain your thinking here? It seems like a lot of things get cleared up to have the id set here: we have a number of methods that were taking both an intake form and an ID, now those can just take the intake form knowing the ID is in there.

Reply:
My thought process is that the initial data should be the very minimum of the application data. It shouldn't have an ID yet because it's "blank".

When a blank form is loaded, the application should generate an id for it.

A potential use case would be a "Clear All Fields" button. When we want to clear a form, ideally, we'd be replacing it with the initial data. Then application would be responsible repopulating its ID. It would feel _wrong_ to overwrite an existing id.

Passing the ID as a separate argument is a way to do it. If we don't want to pass another argument, we can set the id on mount or on save by using some of the Formik methods.

IIRC, `formikProps.setFieldValue('id', uuidv4())`

Reply:
Here's a not-so-great analogy:

I see the initial data as a blank paper form. There's no uniqueness to them until they are given to a person to fill out.

Reply:
Okay! I'll change it to set the ID on mount.